pkg/service/pilot/fg_client_manager.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 The OpenPitrix Authors. All rights reserved.
+// Use of this source code is governed by a Apache license
+// that can be found in the LICENSE file.
+
+package pilot
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"sync"
+
+	"openpitrix.io/openpitrix/pkg/logger"
+	"openpitrix.io/openpitrix/pkg/pb/frontgate"
+	"openpitrix.io/openpitrix/pkg/pb/types"
+)
+
+type FrontgateClientManager struct {
+	clientMap map[string][]*_FGClient
+	sync.Mutex
+}
+
+type _FGClient struct {
```

Comment:
Use camelcase instead of snake case, so this struct should be `fgClient`

Reply:
done